test/e2e-encryption/main_test.go

```diff
@@ -0,0 +1,31 @@
+package e2e_encryption
+
+import (
+	"math/rand"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+	"unsafe"
+)
+
+func TestMain(m *testing.M) {
+	randomizeTestOrder(m)
```

Comment:
wouldn't be better to make the tests `table-driven`, shuffle the table (or just generate random numbers - len(table) and use `t.Run` - https://golang.org/pkg/testing/#T.Run

```
scenarios := []struct {}
order := genRandNumbers(len(scenarios))
for index, _ := range order {
 t.Run(scenarios[index])
}
```

Reply:
I prefer the current approach because it has the least maintenance burden and should be easy to migrate once https://github.com/golang/go/issues/28592 is addressed.

Reply:
Isn't this just a one-year-old proposal? We don't know when and if it will be merged. On the other hand, `table-driven` approach is accepted and well understood.

Reply:
There are multiple issues on the table approach:

1. You have to know that you need to do that in the first place
1. Thus we have to write docs for ^
1. But we have no way to enforce ^
1. If someone adds a new test, we have to make sure they follow ^ otherwise they will have tests that run in the same order
1. `go test -list` does not show subtests so we cannot easily observe if things are being run in random without actually running the tests

I consider this to be a benign use of `reflect` and `unsafe`.  They will `panic` / seg fault if go changes how it lets you mess with this stuff (but this approach has worked for years for me without issue and I suspect it will keep working for a long time).

I want someone to be able to add a test to this suite without having to know anything about how the other tests are run.